library/pkcs7.c

```diff
@@ -541,13 +556,21 @@ int mbedtls_pkcs7_parse_der( mbedtls_pkcs7 *pkcs7, const unsigned char *buf,
     pkcs7->raw.len = buflen;
     end = p + buflen;
 
-    ret = pkcs7_get_content_info_type( &p, end, &pkcs7->content_type_oid );
+    ret = pkcs7_get_content_info_type( &p, end, &end_content_info,
+                                       &pkcs7->content_type_oid );
     if( ret != 0 )
     {
         len = buflen;
         goto try_data;
     }
 
+    /* ensure bufferlength is only pkcs7 data */
```

Comment:
Wording is a bit unclear. Suggest e.g.: "Ensure PKCS7 data uses the exact number of bytes specified in buflen"

Reply:
This requirement should be documented in pkcs7.h - otherwise users may think it's OK to pass a buffer with some empty space at the end.